anghammarad/src/main/scala/com/gu/anghammarad/models/Serialization.scala

```diff
@@ -7,16 +7,27 @@ import io.circe.Decoder.Result
 import io.circe._
 import io.circe.parser._
 
+import scala.collection.JavaConverters._
 import scala.util.{Failure, Success, Try}
 
 
 object Serialization {
-
-    def parseAllNotifications(snsEvent: SNSEvent): Try[List[Notification]] = {
-      ???
+  def parseNotification(snsEvent: SNSEvent): Try[Notification] = {
+    val maybeSns = snsEvent.getRecords.asScala.toList match {
+      case singleRecord :: Nil => Success(singleRecord.getSNS)
+      case _ => Fail(s"Found multiple SNSRecords")
```

Comment:
One very minor point - presumably this will also catch an empty list, so we could make the message a bit more generic...

Reply:
Good point. Probably worth matching empty list separately so we can provide a nice error for each case.